File: src/main/java/com/baidu/algorithm/hashtable/_128_Longest_Consecutive_Sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums is None or len(nums) < 1:
            return 0

        mp = {}
        for x in nums:
            if not mp.has_key(x):
                left = mp.get(x - 1, 0)
                right = mp.get(x + 1, 0)
                mp[x - left] = mp[x + right] = mp[x] = left + right + 1

        return max(mp.itervalues())

File: src/main/java/com/baidu/algorithm/hashtable/_128_Longest_Consecutive_Sequence.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.hashtable;

import java.util.HashSet;
import java.util.Set;

/**
 * _128_Longest_Consecutive_Sequence
 *
 * @author xuhaoran01
 */
public class _128_Longest_Consecutive_Sequence {

    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int res = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int curNum = num, curLen = 1;
                while (set.contains(curNum + 1)) {
                    curNum += 1;
                    curLen += 1;
                }

                res = Math.max(res, curLen);
            }
        }

        return res;
    }
}
